mqtt/mqtt-bridge/src/bridge.rs

```diff
@@ -1,31 +1,387 @@
-use tracing::info;
+use std::{collections::HashMap, marker::PhantomData, str::FromStr, time::Duration};
 
-use crate::settings::ConnectionSettings;
+use anyhow::Result;
+use async_trait::async_trait;
+use mqtt3::{proto::Publication, Event, ReceivedPublication};
+use mqtt_broker::TopicFilter;
+use tracing::{debug, info, warn};
+
+use crate::client::{EventHandler, MqttClient};
+use crate::persist::{memory::InMemoryPersist, Persist};
+use crate::settings::{ConnectionSettings, Credentials, Topic};
+
+const BATCH_SIZE: usize = 10;
 
 /// Bridge implementation that connects to local broker and remote broker and handles messages flow
 pub struct Bridge {
+    system_address: String,
+    device_id: String,
     connection_settings: ConnectionSettings,
+    forwards_map: HashMap<String, Topic>,
+    subs_map: HashMap<String, Topic>,
 }
 
 impl Bridge {
-    pub fn new(connection_settings: ConnectionSettings) -> Self {
+    pub fn new(
+        system_address: String,
+        device_id: String,
+        connection_settings: ConnectionSettings,
+    ) -> Self {
+        let forwards_map: HashMap<String, Topic> = connection_settings
```

Comment:
collect combinator wants to know a type of resulting collection, not items. so we can omit that like
```suggestion
        let forwards_map: HashMap<_> = connection_settings
```
or even 
```suggestion
        let forwards_map = connection_settings
```
because you construct Bridge and HashMap required there